src/QMCTools/QMCFiniteSize/QMCFiniteSize.cpp

```diff
@@ -321,7 +321,10 @@ NUBspline_1d_d* QMCFiniteSize::spline_clamped(vector<RealType>& grid,
   xBC.rCode = DERIV1;
   //hack to interface to NUgrid stuff in double prec for MIXED build
   vector<FullPrecRealType> vals_fp(vals.begin(), vals.end());
-  return create_NUBspline_1d_d(grid1d, xBC, vals_fp.data());
+  auto* nubspline_1d_d = create_NUBspline_1d_d(grid1d, xBC, vals_fp.data());
```

Comment:
Cloud use unique_ptr as  https://github.com/QMCPACK/qmcpack/pull/3191/files

destroy_Bspline should work for NUBspline as well.
I don't think destroy_NUBspline_1d_d is needed.

Reply:
Please leave as the matched delete call - this is more consistent with how C-like einspline works. That is, unless there are any known problems.

Reply:
destroy_Bspline is the way einspline library provides to destroy the object. We should to stick to it or change all the library API instead of doing it ad-hoc.

In addition, I do noticed `destroy_NUBspline` seems mussing and `free(spline)`

Reply:
Thanks for the explanation. Agreed.

Reply:
@ye-luo @prckent thanks for your review. The leak still shows with `destroy_Bspline`, hence the need for the extra deallocation function. As a side note, we can always think about moving third party implementations to a different location other than `src` if it's a separate product altogether, thoughts?

Reply:
`auto integrand =
      std::unique_ptr<NUBspline_1d_d, void (*)(void*)>{spline_clamped(nonunigrid1d, k2vksk, 0.0, 0.0), destroy_Bspline};`

``` Direct leak of 56 byte(s) in 1 object(s) allocated from:
33:     #0 0x4d07c3 in __interceptor_malloc (/home/wgodoy/workspace/qmcpack/build/tests/bin/test_tools+0x4d07c3)
33:     #1 0x90368c in create_NUBspline_1d_d /home/wgodoy/workspace/qmcpack/build/../src/einspline/nubspline_create.c:547:28
```

Reply:
> In addition, I noticed destroy_NUBspline seems missing a free(spline)

Did you try to fix this?

Reply:
> destroy_Bspline should work for NUBspline as well.

@ye-luo so it should be `destroy_NUBspline` not `destroy_Bspline` . Let me try 'cause there might be some casting needed.

Reply:
I think I get it now. I see `destroy_Bspline` calls `destroy_NUBspline` implicitly. Thanks.